src/main/java/seedu/address/ui/AnalyticsPanel.java

```diff
@@ -5,34 +5,99 @@
+    private Label impactIndex;
+
+    @FXML
+    private Label urgencyIndex;
 
     /**
      * Creates a {@code AnalyticsPanel} with the given {@code ObjectProperty}.
      */
-    public AnalyticsPanel(ObjectProperty<Analytics> analytics) {
+    public AnalyticsPanel(ObjectProperty<DashboardData> dashboardData) {
         super(FXML);
-        pieChart.setData(FXCollections.observableArrayList());
-        analytics.addListener((observable, oldValue, newValue) -> {
+        initializeCharts();
+        reliabilityChart.setData(FXCollections.observableArrayList());
+        dashboardData.addListener((observable, oldValue, newValue) -> {
             updateChart(newValue);
         });
     }
 
-    private void updateChart(Analytics analytics) {
-        ObservableList<PieChart.Data> pieChartData = FXCollections.observableArrayList(
-                new PieChart.Data("Active Loans", analytics.getNumActiveLoans()),
-                new PieChart.Data("Overdue Loans", analytics.getNumOverdueLoans())
-        );
-        pieChart.setData(pieChartData);
+    private void initializeCharts() {
+        reliabilityChart.setStartAngle(90);
+        impactChart.setStartAngle(90);
+        urgencyChart.setStartAngle(90);
+        reliabilityChart.setLabelsVisible(false);
+        impactChart.setLabelsVisible(false);
+        urgencyChart.setLabelsVisible(false);
+        reliabilityChart.setLegendVisible(false);
+        impactChart.setLegendVisible(false);
+        urgencyChart.setLegendVisible(false);
+    }
+
+    private void updateChart(DashboardData data) {
+        Analytics analytics = data.getAnalytics();
+        if (analytics.getNumActiveLoans() == 0) {
+            reliabilityIndex.setText("No active loans to analyze");
+            reliabilityChart.setVisible(false);
+        } else {
+            reliabilityChart.setVisible(true);
+            ObservableList<PieChart.Data> reliabilityData = FXCollections.observableArrayList(
+                    new PieChart.Data("Reliability Index", analytics.getPropOverdueLoans()),
+                    new PieChart.Data("", 1 - analytics.getPropOverdueLoans())
+            );
+            reliabilityChart.setData(reliabilityData);
+            reliabilityIndex.setText(String.format("%.2f", (1 - analytics.getPropOverdueLoans()) * 100) + "%");
+        }
+
+
+        if (data.getMaxLoanValue() == 0) {
+            impactIndex.setText("No loans to analyze");
+            impactChart.setVisible(false);
+        } else {
+            impactChart.setVisible(true);
+            ObservableList<PieChart.Data> impactData = FXCollections.observableArrayList(
+                    new PieChart.Data("Impact Index", data.getImpactIndex()),
+                    new PieChart.Data("", 1 - data.getImpactIndex())
+            );
+            impactChart.setData(impactData);
+            impactIndex.setText(String.format("%.2f", data.getImpactIndex() * 100) + "%");
+        }
+        if (data.getUrgencyIndex() == null) {
```

Comment:
There's probably a better to handle these null conditions. Something to consider for future iterations.